PointProcess class with virtual functions but non-virtual destructor

When I build `spatstat` with `-Wnon-virtual-dtor` I get warnings:

```
R CMD INSTALL --preclean --configure-vars="CXXFLAGS=-g -Wnon-virtual-dtor -Werror" spatstat



g++ -std=gnu++11 -I"/usr/local/lib/R/include" -DNDEBUG   -I/usr/local/include  -Wnon-virtual-dtor -fpic  -g -O2  -c Perfect.cc -o Perfect.o
Perfect.cc:326:7: warning: ‘class PointProcess’ has virtual functions and accessible non-virtual destructor [-Wnon-virtual-dtor]
 class PointProcess {
       ^~~~~~~~~~~~
In file included from Perfect.cc:844:0:
PerfectStrauss.h:5:7: warning: base class ‘class PointProcess’ has accessible non-virtual destructor [-Wnon-virtual-dtor]
 class StraussProcess : public PointProcess {
       ^~~~~~~~~~~~~~
PerfectStrauss.h:5:7: warning: ‘class StraussProcess’ has virtual functions and accessible non-virtual destructor [-Wnon-virtual-dtor]
In file included from Perfect.cc:845:0:
PerfectStraussHard.h:5:7: warning: base class ‘class PointProcess’ has accessible non-virtual destructor [-Wnon-virtual-dtor]
 class StraussHardProcess : public PointProcess {
       ^~~~~~~~~~~~~~~~~~
PerfectStraussHard.h:5:7: warning: ‘class StraussHardProcess’ has virtual functions and accessible non-virtual destructor [-Wnon-virtual-dtor]
In file included from Perfect.cc:846:0:
PerfectHardcore.h:5:7: warning: base class ‘class PointProcess’ has accessible non-virtual destructor [-Wnon-virtual-dtor]
 class HardcoreProcess : public PointProcess {
       ^~~~~~~~~~~~~~~
PerfectHardcore.h:5:7: warning: ‘class HardcoreProcess’ has virtual functions and accessible non-virtual destructor [-Wnon-virtual-dtor]
In file included from Perfect.cc:847:0:
PerfectDiggleGratton.h:5:7: warning: base class ‘class PointProcess’ has accessible non-virtual destructor [-Wnon-virtual-dtor]
 class DiggleGrattonProcess : public PointProcess {
       ^~~~~~~~~~~~~~~~~~~~
PerfectDiggleGratton.h:5:7: warning: ‘class DiggleGrattonProcess’ has virtual functions and accessible non-virtual destructor [-Wnon-virtual-dtor]
In file included from Perfect.cc:848:0:
PerfectDGS.h:9:7: warning: base class ‘class PointProcess’ has accessible non-virtual destructor [-Wnon-virtual-dtor]
 class DgsProcess : public PointProcess {
       ^~~~~~~~~~
PerfectDGS.h:9:7: warning: ‘class DgsProcess’ has virtual functions and accessible non-virtual destructor [-Wnon-virtual-dtor]
In file included from Perfect.cc:849:0:
PerfectPenttinen.h:5:7: warning: base class ‘class PointProcess’ has accessible non-virtual destructor [-Wnon-virtual-dtor]
 class PenttProcess : public PointProcess {
       ^~~~~~~~~~~~
PerfectPenttinen.h:5:7: warning: ‘class PenttProcess’ has virtual functions and accessible non-virtual destructor [-Wnon-virtual-dtor]
```

This can be resolved with a quick fix to the `PointProcess` class definition:

```
~PointProcess(){}

// becomes
virtual ~PointProcess(){}
```

--- a/src/Perfect.cc
+++ b/src/Perfect.cc
@@ -330,7 +330,7 @@ class PointProcess {
     Xmin = xmin; Xmax = xmax;
     Ymin = ymin; Ymax = ymax;
   }
-  ~PointProcess(){}
+  virtual ~PointProcess(){}
   virtual void NewEvent(double *x, double *y, char *InWindow)=0;
   virtual void GeneratePoisson(Point *headPoint, 
 			       long int *GeneratedPoints,

